core/src/main/java/org/apache/accumulo/core/gc/ReferenceDirectory.java

```diff
@@ -21,13 +21,22 @@
 import org.apache.accumulo.core.data.TableId;
 
 /**
- * Part of the Tablet File path that is definitely a directory.
+ * A GC reference to a Tablet directory, like t-0003.
  */
-public class ReferenceDirectory extends Reference {
-  public final String tabletDir; // t-0003
+public class ReferenceDirectory extends ReferenceFile {
+  private final String tabletDir; // t-0003
 
   public ReferenceDirectory(TableId tableId, String dirName) {
     super(tableId, dirName);
```

Comment:
Would some validation make sense for this new class?  Something like the following as an example, but not sure its correct.

```suggestion
  private static validateDirName(String dirName) {
     checkArgument(dirName.matches("[0-9A-Za-z]+")));
  }
 
  public ReferenceDirectory(TableId tableId, String dirName) {
    super(tableId, validateDirName(dirName));
```

Reply:
We do have a method to check the dirName, but I think it could be improved:
https://github.com/apache/accumulo/blob/cf57e343de77f416de10c22e0ed05ef37f433b36/core/src/main/java/org/apache/accumulo/core/metadata/schema/MetadataSchema.java#L199

Reply:
I think this needs an underscore and "-". We have the value "default_tablet" and tablet directories have a dash like:
<pre>
2;9 srv:dir []	t-000009x
2< srv:dir []	default_tablet
</pre>